data/pigui/modules/radar.lua

```diff
@@ -95,144 +164,212 @@ local function display2DRadar(cntr, size)
+-- between frames. We are trying to ensure that a mouse right-click started and
+-- finished inside the radar area in order to trigger the popup.
 local click_on_radar = false
+
 -- display either the 3D or the 2D radar, show a popup on right click to select
 local function displayRadar()
 	if ui.optionsWindow.isOpen or Game.CurrentView() ~= "world" then return end
-	player = Game.player
+	player = player or Game.player
 
 	-- only display if there actually *is* a radar installed
 	local equipped_radar = player:GetComponent("EquipSet"):GetInstalledOfType("sensor.radar")
-	if #equipped_radar > 0 then
-
-		local size = ui.reticuleCircleRadius * 0.66
-		local cntr = Vector2(ui.screenWidth / 2, ui.screenHeight - size - 4)
+	-- TODO: get ship radar capability and determine functionality based on level of radar installed
+	if #equipped_radar  == 0 then return end
+
+	local instrument = shouldDisplay2DRadar and radar2d or radar3d
+	local center = Vector2(ui.screenWidth / 2, ui.screenHeight - radar2d.size - SCREEN_BORDER)
+	local zoom = 0
+	local toggle_radar = false
+
+	-- Handle keyboard
+	-- TODO: Convert to axis?
+	if keys.radar_zoom_in:IsJustActive() then
+		zoom = 1
+	elseif keys.radar_zoom_out:IsJustActive() then
+		zoom = -1
+	end
+	if keys.radar_reset:IsJustActive() then
+		zoom = 0
+		instrument:resetZoom()
+	end
+	if keys.radar_toggle_mode:IsJustActive() then
+		toggle_radar = true
+	end
 
-		local mp = ui.getMousePos()
-		if (mp - cntr):length() > size then
-			click_on_radar = false
-		end
-		if (mp - cntr):length() < size then
-			if ui.isMouseClicked(1) then
-				click_on_radar = true
-			end
-			if click_on_radar and ui.isMouseReleased(1) then
-				ui.openPopup("radarselector")
-			end
-			local wheel = ui.getMouseWheel()
-			if wheel > 0 then
-				current_radar_size = math.max(current_radar_size / 10, MIN_RADAR_SIZE)
-			elseif wheel < 0 then
-				current_radar_size = math.min(current_radar_size * 10, MAX_RADAR_SIZE)
-			end
-		end
+	-- Handle mouse if it is in the radar area
+	local mp = ui.getMousePos()
+	-- TODO: adjust properly for 3D radar; bit more challenging as its an ellipse
+	if (mp - center):length() < radar2d.getRadius(radar2d) then
 		ui.popup("radarselector", function()
 			if ui.selectable(lui.HUD_2D_RADAR, shouldDisplay2DRadar, {}) then
-				Event.Queue('onChangeMFD', 'radar')
+				toggle_radar = true
 			end
 			if ui.selectable(lui.HUD_3D_RADAR, not shouldDisplay2DRadar, {}) then
-				Event.Queue('onChangeMFD', 'scanner')
+				toggle_radar = true
 			end
 		end)
-		if shouldDisplay2DRadar then
-			display2DRadar(cntr, size)
-		else
-			display3DRadar(cntr, Vector2(ui.reticuleCircleRadius * 1.8, size * 2))
+
+		if ui.isMouseClicked(1) then
+			click_on_radar = true
+		end
+		if not toggle_radar and click_on_radar and ui.isMouseReleased(1) then
+			ui.openPopup("radarselector")
+		end
+		-- TODO: figure out how to "capture" the mouse wheel to prevent
+		-- the game engine from using it to also zoom the viewport
```

Comment:
For this, you'll want to push an InputFrame (`Input.CreateInputFrame("RadarScroll")`) when the player starts hovering the "scrollable area" and pop it when the mouse leaves. That InputFrame should contain an axis binding with mouse wheel up/down assigned to its positive/negative bindings.

The Input system will then allow that InputFrame to "consume" the mouse wheel motion, and the lower-priority binding to zoom the main view will not be activated.

Reply:
There is no code anywhere which binds an axis to the mousewheel.. there are simply APIs called "getWheel" (or similar). Both C++ and Lua side.  There are axis definitions for joysticks and keyboard pairs. I suspect some additional plumbing is required somewhere? Or I'm misunderstanding how the mouse wheel can be turned into an axis..

Reply:
Ok, fixing the mouse zoom is out-of-scope for this PR.